diesel/src/sqlite/connection/raw.rs

```diff
@@ -292,99 +313,160 @@ extern "C" fn run_aggregator_step_function<ArgsSqlType, RetSqlType, Args, Ret, A
         //   the memory will have a correct alignment.
         //   (Note I(weiznich): would assume that it is aligned correctly, but we
         //    we cannot guarantee it, so better be safe than sorry)
-        let aggregate_context = ffi::sqlite3_aggregate_context(
-            ctx,
-            std::mem::size_of::<OptionalAggregator<A>>() as i32,
-        );
-        let mut aggregate_context = NonNull::new(aggregate_context as *mut OptionalAggregator<A>);
-        let aggregator = match aggregate_context.map(|a| &mut *a.as_ptr()) {
+        ffi::sqlite3_aggregate_context(ctx, std::mem::size_of::<OptionalAggregator<A>>() as i32)
+    };
+    let mut aggregate_context = NonNull::new(aggregate_context as *mut OptionalAggregator<A>);
+    let aggregator = unsafe {
+        match aggregate_context.map(|a| &mut *a.as_ptr()) {
             Some(&mut OptionalAggregator::Some(ref mut agg)) => agg,
             Some(mut a_ptr @ &mut OptionalAggregator::None) => {
                 ptr::write_unaligned(a_ptr as *mut _, OptionalAggregator::Some(A::default()));
                 if let &mut OptionalAggregator::Some(ref mut agg) = a_ptr {
                     agg
                 } else {
-                    unreachable!(
-                        "We've written the aggregator above to that location, it must be there"
-                    )
+                    eprintln!("We've written the aggregator to the aggregate context, but it could not be retrieved");
```

Comment:
Could you improve that error message saying something like: If you ever see this message, please open an issue at the diesel github repository?


(Maybe with an indication on which line this error happened?)